### ISSUE TITLE
Extension webpage cannot receive message

### ISSUE THREAD

**Issue by spcidealacm** [coder/code-server#1860]
Vscode extension can load webpages and send messages to them.
And the webpages can receive these messages through below code:
```
window.addEventListener('message', event => {
   ...
}
```
Related articles:
https://code.visualstudio.com/api/extension-guides/webview#scripts-and-message-passing

But in my test, vscode can make it work meanwhile code-server cannot do that.
![image](https://user-images.githubusercontent.com/8594876/86425438-88390080-bd28-11ea-817a-b4d67de751be.png)

This picture I have let console print message.status. and if receive "connected", it will change page info.
However, the console has shown the extension already sent a message to the webpage. 
![image](https://user-images.githubusercontent.com/8594876/86425971-de5a7380-bd29-11ea-8a8f-07cf96cac14d.png)

The console hasn't shown any info. And the webpage hasn't changed. 
![image](https://user-images.githubusercontent.com/8594876/86426324-bae3f880-bd2a-11ea-9f6e-607e09784854.png)

In the same situation, Vscode can receive information and respond.
![image](https://user-images.githubusercontent.com/8594876/86426373-e2d35c00-bd2a-11ea-9cda-00a589847b9f.png)

My extension is "code map" which has been published on the vscode marketplace. I also want it can work on code-server.
![image](https://user-images.githubusercontent.com/8594876/86426433-04ccde80-bd2b-11ea-9678-78e927900b7d.png)

Please fix this bug or give me a solution :)

**Comment by nhooyr** [comment]
cc @code-asher

**Comment by nhooyr** [comment]
Sorry, looks like we missed this!

**Comment by code-asher** [comment]
@spcidealacm I'm trying to replicate the issue, could you clarify what function needs to be called `postMessage`? Is it the event handler shown in the images you posted?

I tried running the sample webview extension (https://github.com/microsoft/vscode-extension-samples/tree/master/webview-sample) and then I changed the `message` event:

```
    const postMessage = event => {
        const message = event.data; // The json data that the extension sent
        switch (message.command) {
            case 'refactor':
                currentCount = Math.ceil(currentCount * 0.5);
                counter.textContent = currentCount;
                break;
        }
    }
    window.addEventListener('message', postMessage);
```

But this is working for me.

**Comment by nhooyr** [comment]
Going to close as there hasn't been a response in a long time.

**Comment by TheCoryBarker** [comment]
@code-asher I think I'm running into the same issue. To repro:

```
function postMessage(data){
  console.log('something');
}

const otherFunction = (event) => {
  console.log("hi");
}

window.addEventListener("message", otherFunction);
```

`webview.postMessage(data);` fails everytime when there is another function called `postMessage`. So to fix, change `postMessage` to `postMessage2` and it works:

```
function postMessage2(data){
  console.log('something');
}

const otherFunction = (event) => {
  console.log("hi");
}

window.addEventListener("message", otherFunction);
```

**Comment by code-asher** [comment]
Oh I see!  Interesting, I will re-open the issue.

**Comment by code-asher** [comment]
We should test this in Codespaces to see how it behaves.  Possibly this is an upstream issue.

**Comment by code-asher** [comment]
Closing for now, but let me know if you come back to this and whether it reproduces in Codespaces.  I would expect it to be an upstream issue since we have not modified any of this code.

**Comment by edsonfischborn** [comment]
Hello, today I also encountered this issue while developing my extension.

I am using React to simplify event handling in the webview. To send data from the webview to the extension, I created a function called `postMessage` in my component, which worked well for sending data. However, I had no success receiving data with `window.addEventListener('message')`, as the event was never triggered.

After reading this topic and renaming the function, the listener `window.addEventListener('message')` started to be triggered.

After identifying this issue, I decided to check how the React build was being generated and noticed that the `postMessage` function was created at the root of the bundle file using the `var` keyword.

![image](https://github.com/user-attachments/assets/f1b1481c-b8df-4848-8b6d-9d8810b35b4d)

To simulate this situation in Codespaces, I decided to create a simple extension that employs the same concept by implementing the `vscode.WebviewViewProvider` interface and simulating the declaration of `postMessage` within the injected script in the HTML.

![run-demo](https://github.com/user-attachments/assets/780cd448-1684-401b-a0ae-bd31c90f3c56)

`[gff](https://github.com/edsonfischborn/vscode-ext-event-test/blob/main/media/main.js)`
